### PR TITLE
Implement math.isqrt in Rust

### PR DESCRIPTION
**Pull request by ArniDagur**

### CODE REVIEW

File: Lib/test/test_math.py [hunk 1]
```diff
@@ -957,19 +956,20 @@ def testIsqrt(self):
         self.assertIs(type(s), int)
         self.assertEqual(s, 0)
 
-        class IntegerLike(object):
-            def __init__(self, value):
-                self.value = value
+        # TODO: RUSTPYTHON
```
Comment (ArniDagur) [inline Lib/test/test_math.py:959]: Any idea how I can support this non-integer argument?

Comment (youknowone) [reply Lib/test/test_math.py:959]: It seems the function take not only int but also types with `__index__`. Search for `vm.to_index` in code. `bytesinner.rs` and `pystruct.rs` look like similar case.

Comment (ArniDagur) [reply Lib/test/test_math.py]: Thanks :)

**Review by youknowone — APPROVED** [review]

**Comment by youknowone** [comment]
thank you for contributing!

**Pull request merged**